src/main/scala/org/clulab/timenorm/scfg/Evaluator.scala

```diff
@@ -0,0 +1,193 @@
+import org.clulab.timenorm.scfg._
+import scala.util.{Success, Failure}, scala.io.Source
+import java.io._
+import scala.collection.mutable.ListBuffer
+
+object Evaluator {
+  /**
+  This program normalizes timexes and compares the results to their gold
+  standard normalizations
+  */
+
+  def main(lang:String, in_file:String, out_file:String) {
+    /**
+    Enter the language ("es"/"en") and the input and output paths
+
+    Both input and output files are .tsv files with timexes in the 1st column,
+    gold normalization value in the 2nd column and, in the output file, system
+    normalization value in the 3rd column. Timexes from different documents must
+    be separated by newlines, being DCTs the first timexes from each document
+    */
+
+    // Obtain the data of timexes and gold values from the input file
+    val (timex_list, gold_list) = get_content(in_file)
+    // Obtain the normalizations of the timexes.
+    val norm_list = get_normalizations(lang, timex_list)
+    // Compare gold and system normalizations, write the results and get the
+    //sums of timexes and correct normalizations
+    val (sum_gold, sum_norm) = compare_and_write(out_file, timex_list, gold_list, norm_list)
+
+    // Compute number of errors and accuracy
+    var sum_errors = sum_gold - sum_norm
+    var accuracy = sum_norm.toFloat * 100 / sum_gold
+
+    // Print the final statistics
+    println(f"""\n
+                |Number of timexes (also DCTs): $sum_gold%6d
+                |Correct normalizations:        $sum_norm%6d
+                |Incorrect normalizations:      $sum_errors%6d
+                |Accuracy:                      $accuracy%6.2f\n""".stripMargin)
+  }
+
+
+  def get_content(in_file:String) : (List[String], List[String]) = {
+    /**Obtains the content from the input file as timex and value lists*/
+
+    // Turn input file to a list of lines
+    val content = Source.fromFile(in_file).getLines.toList
+    // Obtain the standard line length from the first DCT
+    val std_line_length = content(0).split("\t").length
+
+    val timex_list = ListBuffer[String]()
+    val gold_list = ListBuffer[String]()
+
+    for (line <- content) {
+      // If line is not a doc separator (indicated by empty string):
+      if (line != "") {
+        // If line length equals the standard, get the timex and its gold value
+        if (line.split("\t").length == std_line_length) {
+          timex_list += line.split("\t").head
+          gold_list += line.split("\t").last
+        }
+        // If this is a detected timex absent in the evaluation corpus, get the
+        // timex but append "" as gold normalization
+        else {
+          timex_list += line.split("\t").head
+          gold_list += "-"
+        }
+      }
+      // Otherwise, add empty strings to mark end of document timexes
+      else {
+        timex_list += ""
+        gold_list += ""
+      }
+    }
+    return (timex_list.toList, gold_list.toList)

```

Comment:
It looks like we're not writing `return`.  Can they be removed throughout?

Reply:
I appreciate any improvement for a better performance :)